lib/resourcebuilder/apps.go

```diff
@@ -77,6 +80,18 @@ func waitForDeploymentCompletion(ctx context.Context, client appsclientv1.Deploy
 		klog.V(4).Infof("Deployment %s is not ready. status: (replicas: %d, updated: %d, ready: %d, unavailable: %d)", d.Name, d.Status.Replicas, d.Status.UpdatedReplicas, d.Status.ReadyReplicas, d.Status.UnavailableReplicas)
 		return false, nil
 	}, ctx.Done())
+
+	if err == wait.ErrWaitTimeout {
+		message := fmt.Sprintf("Deployment %s failed to become ready in time.", d.Name)
+		if d != nil {
+			message +=  fmt.Sprintf("status: (replicas: %d, updated: %d, ready: %d, unavailable: %d)", d.Status.Replicas, d.Status.UpdatedReplicas, d.Status.ReadyReplicas, d.Status.UnavailableReplicas)
+		}
+		klog.Warningf(message)
+
+		return errors.New(message)
```

Comment:
unfortunately to do this you need to return an update error and have handling that indicates the why.  you can't use errors.New()